src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -326,18 +362,18 @@ private UpdateLevel GetUpdateLevel(NuGetVersion resolvedVersion, NuGetVersion la
         /// <param name="providers">The providers to use when looking at sources</param>
         /// <param name="packagesVersionsDict">A reference to the unique packages in the project
         /// to be able to handle different sources having different latest versions</param>
-        /// <returns>An updated pacakge with the highest version at a single source</returns>
+        /// <returns>An updated package with the highest version at a single source</returns>
         private async Task GetLatestVersionPerSourceAsync(
             PackageSource packageSource,
             ListPackageArgs listPackageArgs,
             string package,
             IEnumerable<Lazy<INuGetResourceProvider>> providers,
-            Dictionary<string, IList<NuGetVersion>> packagesVersionsDict)
+            Dictionary<string, IList<NuGetVersionWithDeprecationInfo>> packagesVersionsDict)
         {
             var sourceRepository = Repository.CreateSource(providers, packageSource, FeedType.Undefined);
             var dependencyInfoResource = await sourceRepository.GetResourceAsync<MetadataResource>(listPackageArgs.CancellationToken);
```

Comment:
Can you please rename this to metadataResource? 
It got me so many times :D 
The dependencyinforesource is different, lol.

Reply:
Done.